### ISSUE TITLE
Multi module project with quarkus main, restarts on change in not dependent project

### ISSUE THREAD

**Issue by andrzejszywala** [quarkusio/quarkus#13298]
**Describe the bug**
I've created maven project 
[example.zip](https://github.com/quarkusio/quarkus/files/5542077/example.zip)
 with 3 modules:

- server - quarkus appliaction with main class ServerMain. It has shared module in dependencies
- shared - classes shared  between client and server
- client - client appliaction. It has shared module in dependencies

After importing project to eclipse and run main method from ServerMain class. Any changes in client module causes redeploy of quarkus appliaction.

**Expected behavior**
Because client module is not in dependencies of server module. Changes in client module shouldn't cause redeploy of quarkus application

**Actual behavior**
After change in ClientMain class quarkus application detects that source files has been changed and redeploys application.
```
2020-11-15 10:15:47,503 INFO  [io.qua.dep.dev.RuntimeUpdatesProcessor] (vert.x-worker-thread-1) Changed source files detected, recompiling [D:\praca\quarkus\example\client\src\main\java\com\acme\ClientMain.java]
2020-11-15 10:15:48,469 INFO  [io.quarkus] (Quarkus Main Thread) Quarkus stopped in 0.040s
Staring... 
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2020-11-15 10:15:49,144 INFO  [io.quarkus] (Quarkus Main Thread) Quarkus 1.9.2.Final on JVM started in 0.666s. Listening on: http://0.0.0.0:8080
2020-11-15 10:15:49,144 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2020-11-15 10:15:49,144 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, resteasy, resteasy-jackson]
2020-11-15 10:15:49,145 INFO  [io.qua.dep.dev.RuntimeUpdatesProcessor] (vert.x-worker-thread-1) Hot replace total time: 1.646s 
```

**To Reproduce**

1. In Eclipse select File->Import->Existing Maven Projects
2. Select folder with attached application
3. In Project explorer select ServerMain.java file
4. Right click, select Run As -> Java appliaction
5. In Project explorer select ClientMain.java
6. Right click, select Run As -> Java appliaction
7. Uncomment line 16 in ClientMain.java
8. Save file
9. Run ClientMain class

**Environment (please complete the following information):**
 - Microsoft Windows [Version 10.0.18363.1198]
 - openjdk version "14.0.2" 2020-07-14
   OpenJDK Runtime Environment (build 14.0.2+12-46)
   OpenJDK 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)
 - Quarkus version 1.9.2.Final
 - Build tool Eclipse 2020-09(4.17.0)

**Comment by famod** [comment]
I was curious and so I tried it without Eclipse and with the following steps I could *not* reproduce the issue:
- `mvn clean install -DskipTests`
- `mvn quarkus:dev -f server`
- `mvn exec:java -Dexec.mainClass='com.acme.ClientMain' -f client`
- modify `ClientMain`
- `mvn compile exec:java -Dexec.mainClass="com.acme.ClientMain" -f client`

@andrzejszywala Can you check please? I you also cannot reproduce it this way then it might have something to do with Eclipse.

**Comment by andrzejszywala** [comment]
Yes from maven it works just fine. I think the problem is in the ```io.quarkus.deployment.dev.IDEDevModeMain``` class. In this code fragment
```
if (BuildToolHelper.isMavenProject(appClasses)) {
	LocalProject project = LocalProject.loadWorkspace(appClasses);
	DevModeContext.ModuleInfo root = toModule(project);
	devModeContext.setApplicationRoot(root);
	for (Map.Entry<AppArtifactKey, LocalProject> module : project.getWorkspace().getProjects().entrySet()) {
		if (module.getKey().equals(project.getKey())) {
			continue;
		}
		devModeContext.getAdditionalModules().add(toModule(module.getValue()));
	}
}
```
to the devModeContext.additionalModules shouldn't be added modules which are not in dependencies of quarkus appliaction.
After execution of this loop
```
for (Map.Entry<AppArtifactKey, LocalProject> module : project.getWorkspace().getProjects().entrySet()) {
```
devModeContext.additionalModules looks like the following screen
![screen1](https://user-images.githubusercontent.com/10794859/99488341-3952d700-2968-11eb-9e6a-48800bea8aa9.png)
and I think that client module shouldn't be there.

As a workaround I can change parent pom.xml and move client module to separate profile. After importing all modules to Eclipse it works fine
```
<modules>
	<module>shared</module>
        <module>server</module>
</modules>

<profiles>
	<profile>
		<id>production</id>
		<modules>
			<module>client</module>
		</modules>
	</profile>
</profiles>
```
and only when I building on CI I have to add profile -Pproduction.
